tools/roslaunch/src/roslaunch/xmlloader.py

```diff
@@ -235,9 +235,10 @@ def _rosparam_tag(self, tag, context, ros_config, verbose=True):
             # load is the default command            
             cmd = cmd or 'load'
             value = _get_text(tag)
+            subst_function = None
             if subst_value:
-                value = self.resolve_args(value, context)
-            self.load_rosparam(context, ros_config, cmd, param, file, value, verbose=verbose)
+                subst_function = lambda x: self.resolve_args(x, context)
+            self.load_rosparam(context, ros_config, cmd, param, file, value, verbose=verbose, subst_function=subst_function)
```

Comment:
Without the patch even a plain `value` (no `file`) was being substituted when `subst_value` was set. After this change substitution seems to only happen on the file content. Is that intentional? This seems to change the existing behavior which would be undesired.

Reply:
No, I am not changing default behavior, the code that does the change is moved to 
https://github.com/nano-meter/ros_comm/blob/a7ed7d8c003f52757b75121ee021b578b4b33a07/tools/roslaunch/src/roslaunch/loader.py#L405

This way both value and file content substitution is handled the same way. 
I also created test for file and value substitution, so if I introduced undesired changes, those probably would have caught it. If you see a problem, please tell me so I could fix it.

Reply:
Thanks for clarifying. I see now this is only relevant when `cmd` is `load` where it is handles in the function. The other commands don't mind about `value` so it is ok to not substitute it in these cases.